ActiveLabel/ActiveLabel.swift

```diff
@@ -139,22 +148,141 @@ typealias ElementTuple = (range: NSRange, element: ActiveElement, type: ActiveTy
+            self.lastCopyMenuRect != rect else {
+            return
+        }
+        self.lastCopyMenuRect = rect
+        self.showCopyMenu(rect: rect, sender: sender)
+        self.copyLink = url.description
+    }
+    
+    private func selectedLinkRectangle(link: String, touchPoint: CGPoint) -> CGRect? {
+        guard let text = self.text else {
+            return nil
+        }
+        let linkRanges = self.processLinkRanges(text: text, link: link)
+        for linkRange in linkRanges {
+            let characterRange = NSRange(linkRange, in: text)
+            let boundRect = processBoundRect(forCharacterRange: characterRange)
+            let correctBoundRect = CGRect(x: boundRect.minX, y: boundRect.minY + self.heightCorrection, width: boundRect.width, height: boundRect.height)
+            guard !correctBoundRect.contains(touchPoint) else {
+                self.highlightLink(range: characterRange)
+                return correctBoundRect
+            }
+        }
+        return nil
+    }
+    
+    private func processLinkRanges(text: String, link: String) -> [Range<String.Index>] {
+        var correctLink = link
+        if let maxLength = self.urlMaximumLength,
+            correctLink.count > maxLength {
+            correctLink = correctLink.trim(to: maxLength)
+        }
+        var linkRanges = text.ranges(of: correctLink)
+        guard let range = correctLink.range(of: "://") else {
+            return linkRanges
+        }
+        let linkRange = range.upperBound..<correctLink.endIndex
+        let clippedLink = String(correctLink[linkRange])
+        let clippedLinkRanges = text.ranges(of: clippedLink)
+        for clippedLinkRange in clippedLinkRanges {
+            linkRanges.append(clippedLinkRange)
+        }
+        return linkRanges
+    }
+    
+    private func processBoundRect(forCharacterRange range: NSRange) -> CGRect {
+        var glyphRange = NSRange(location: 0, length: textStorage.length)
+        self.layoutManager.characterRange(forGlyphRange: range, actualGlyphRange: &glyphRange)
+        let boundingRect = layoutManager.boundingRect(forGlyphRange: glyphRange, in: textContainer)
+        return boundingRect
+    }
+    
+    private func showCopyMenu(rect: CGRect, sender: UILongPressGestureRecognizer) {
+        guard let responder = sender.view, responder.becomeFirstResponder() else {
+            self.copyLink = nil
+            return
+        }
+        self.menuController.arrowDirection = .default
+        self.menuController.setTargetRect(rect, in: responder)
+        self.menuController.setMenuVisible(true, animated: true)
+    }
+    
+    private func hideCopyMenu() {
+        self.lastCopyMenuRect = .zero
+        self.copyLink = nil
+        self.menuController.setMenuVisible(false, animated: true)
+    }
+    
+    private func highlightLink(range: NSRange) {
+        self.textStorage.removeAttribute(.backgroundColor, range: NSRange(location: 0, length: self.textStorage.length))
```

Comment:
if string at this range already had .backgroundColor attribute it will be removed and never restored